8_VideoGameStore.ts

```diff
@@ -0,0 +1,106 @@
+// Jaime Restrepo
+// 8. More OOP concepts.
+// Video Games Store.
+
+
+class VideoGame {
+
+    title: string;
+    developer: string;
+    availableCopies: number;
+    price: number;
+    copiesSold: number;
+
+    static availableGames: string[];
+    static soldOutGames: string[];
+    static totalSold: number;
+
+    constructor(title: string, developer: string, availableCopies: number, price: number, copiesSold: number = 0){
+        if (availableCopies <= 0) {
+            throw new Error("The number of available copies should be greather than 0");
+        }
+        if (price <= 0) {
+            throw new Error("The price should be greather than 0");
+        }
+        this.title = title;
+        this.developer = developer;
+        this.availableCopies = availableCopies;
+        this.price = price;
+        this.copiesSold = copiesSold;
+
+        VideoGame.availableGames.push(this.title);
+    }
+
+    isAvailable(): boolean {
+        return this.availableCopies > 0 ? true : false;
+    }
+
+    sell(numberOfCopiesParameter?: number): number {
+
+        let transactionValue: number = 0;
+        let numberOfCopies: number;
+
+        if (!this.isAvailable()){
+            throw new Error(`${this.title} is not available.`);
+        } 
```

Comment:
Here we should update the isAvailable method to be able to check if it has more than a certain amount of copies.
I mean, if we're trying to sell 3 games, but there is one, the algorithm 'will be successful', but it should throw this error.

It could be an optional parameter since the second part where this method is used looks ok as it is.